app/page.tsx

```diff
@@ -8,37 +8,126 @@ import {
   getMarkets,
 } from "@/queries/omen";
 import { CardMarket, LoadingCardMarket } from "@/app/components";
+import {
+  Button,
+  Icon,
+  Input,
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "swapr-ui";
+import { useState } from "react";
+import { useShowClientUI } from "../hooks/useShowClientUI";
+
+type Option = {
+  name: string;
+  orderBy: FixedProductMarketMaker_OrderBy;
+};
+
+const filterOptions = [
+  {
+    name: "Newest",
+    orderBy: FixedProductMarketMaker_OrderBy.CreationTimestamp,
+  },
+  {
+    name: "Highest volume",
+    orderBy: FixedProductMarketMaker_OrderBy.CollateralVolume,
+  },
+];
+
+const ITEMS_PER_PAGE = 12;
 
 export default function HomePage() {
-  const { data: markets, isLoading } = useQuery({
-    queryKey: ["getMarkets"],
+  const showClientUI = useShowClientUI();
+  const [selectedOption, setSelectedOption] = useState<Option>(
+    filterOptions[0]
+  );
+  const [search, setSearch] = useState("");
+
+  const { data, isLoading } = useQuery({
+    queryKey: ["getMarkets", search, selectedOption.orderBy],
     queryFn: async () =>
       getMarkets({
-        first: 10,
+        first: ITEMS_PER_PAGE,
         skip: 0,
-        orderBy: FixedProductMarketMaker_OrderBy.CreationTimestamp,
+        orderBy: selectedOption.orderBy,
         orderDirection: OrderDirection.Desc,
+        title_contains_nocase: search,
       }),
   });
```

Comment:
WDYT about adding a debounce here (for 500ms or 700ms) to reduce the amount of calls we do to query data?